### PR TITLE
Rename RequestInfo to StreamInfo

### LINKED ISSUE #4500: Cleanup: rename RequestInfo
In #4454 we decided that `RequestInfo` should be renamed to `StreamInfo` or similar, or that separate subclasses of it should be created for the Connection vs Http-Request uses of it.

### PR DESCRIPTION
**Pull request by venilnoronha**
*Description*:
This PR renames `RequestInfo` to `StreamInfo` as discussed in [this comment thread](https://github.com/envoyproxy/envoy/pull/4454#discussion_r219324177) and #4500. 

Signed-off-by: Venil Noronha <veniln@vmware.com>

*Risk Level*: Low
*Testing*: Existing tests suffice
*Docs Changes*: N/A
*Release Notes*: Added a line about the change
Fixes #4500.

/cc @ggreenway

### CODE REVIEW

**Comment by mattklein123** [comment]
OK sounds good to hold for 1.8.0. I think we should try to get 1.8.0 out by the end of next week.

**Comment by venilnoronha** [comment]
not stale.

File: docs/root/intro/version_history.rst [hunk 1]
```diff
@@ -3,6 +3,8 @@ Version history
 
 1.9.0 (pending)
 ===============
+* Renamed the `RequestInfo` namespace to `StreamInfo` to better match its
```
Comment (ramaraochavali) [inline docs/root/intro/version_history.rst]: @venilnoronha nit: could you please prefix this item with sub system name as documented in https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md#release-notes?

Comment (venilnoronha) [reply docs/root/intro/version_history.rst]: Addressed via https://github.com/envoyproxy/envoy/pull/4609/files#diff-d8d3e33358b55e6c0466dd1bb5f40c79R6 in #4609.

Comment (ramaraochavali) [reply docs/root/intro/version_history.rst]: Thank you :-)

**Pull request merged**